### DISCUSSION TITLE
Segmentation fault

### DISCUSSION THREAD

**Discussion by rushitkhunt1996** [pimcore/pimcore#17212] [Q&A / Support]
Hello Everyone,

We are facing a segmentation fault issue while using [Azure Storage](https://github.com/orgs/pimcore/discussions/12879#discussioncomment-3463963) for Assets and AppDynamic(php-agent).
Below is the code.

```
    try {

            // First block
            $filename =  'aaa'.date("Y-m-d-His") . '.csv';
            $logFile = PIMCORE_PROJECT_ROOT . '/public/var/files/aaa.csv';
            $folder = \Pimcore\Model\Asset\Service::createFolderByPath('/Rushit');
            $assetObj = new \Pimcore\Model\Asset();
            $assetObj->setKey(\Pimcore\File::getValidFilename($filename));
            $assetObj->setFilename(\Pimcore\File::getValidFilename($filename));
            $assetObj->setData(file_get_contents($logFile));
            $assetObj->setParent($folder);
            $assetObj->save();
           
            // Second Block
            $jobId = 101;
            $JobObj1 = DataObject\AssetJob::getById($jobId);
           
           $versions = $JobObj1->getVersions();
           $countVersion = count($versions);
           $lastVersion = $versions[$countVersion - 1];
           $JobObj = isset($lastVersion) ? $lastVersion->getData() : $JobObj1;  // Issue with getData

            $JobObj->setJobStatus('completed');
            $JobObj->setJobDescription('completed');
            $endTime = Carbon::now();
            $JobObj->setJobEndTime($endTime);
            $JobObj->setJobLogFile($assetObj);
            $JobObj->save();
            
        } catch (\Error $exu) {
            dd($exu);
        } catch (\Exception $e) {
            dd($e);
        }
               
        dd("cool");
        return true;
```

This script we are executing with a custom command, and we are not able to see any error on the log file as well not in any catch area.
Once we execute this script via command we can see the last dd("cool"); on the terminal with a Segmentation Fault error. see screenshot

![image](https://github.com/pimcore/pimcore/assets/60934941/ac8386fd-646b-4924-8409-9fd2d6ac13f2)

**When we are facing this issue**
1. Azure Storage Enabled & AppDynamic(php-agent) Enabled
2. From the code First block and Second block is active

**When we are not facing an issue**
1. Azure Storage Enabled and AppDynamic(php-agent) Disabled
2. Azure Storage Disabled and AppDynamic(php-agent) Enabled
3. From code any one block(First Block | Second Block) is commented

**Comment by leuchtdiode** [comment]
Hi,

we are facing the same issue, but only if we change the `bin/console` file to the new format where the application startup is performed in a function which are getting returned by the console file (see latest version in skeleton). We didn't change this file until we updated to 11.3.0, which awaits now a callable as it looks like.

We can reproduce the error if we are loading about 277 data objects from a folder (via `$listing->load()`). I debugged the loop in the internal class (`Pimcore\Model\DataObject\Listing::load` method). After the 226th entry loaded by `DataObject::getById` -> line `Pimcore\Model\DataObject\Listing:67`), the script fails with "Segmentation fault (core dumped)".
Memory does not seem to be an issue, as the usage is only at ~ 16 MB after this 226th entry.

When I load data objects with the same logic in batches (offset: 0, limit: 100 ... offset: 100, limit: 100 ... offset: 200, limit: 100) I can retrieve all 227 data objects without any problem.

Anyone with an idea? Maybe there is a memory issue within the anonymous function which getting returned from `bin\console`?

Our PHP version is 8.2.20.

BR,
Alex

**Comment by leuchtdiode** [comment]
@rushitkhunt1996 We found the problem. Disabling xdebug solved the issue for us.

**Reply by jdreesen** [reply]
We also experienced issues with `xdebug 3.3.x` and Pimcore, so we are currently working with `xdebug 3.2.2`, which works well for us.
